packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -0,0 +1,329 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { authentication, FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace externalAppAuthentication {
+  /*********** BEGIN REQUEST TYPE ************/
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type OriginalRequestInfo = IQueryMessageExtensionRequest | IActionExecuteInvokeRequest;
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IQueryMessageExtensionRequest {
+    requestType: OriginalRequestType.QueryMessageExtensionRequest;
+    commandId: string;
```

Comment:
>string [](http://example.com/codeflow?start=15&length=6)

Just pretend I put a comment on practically every string and said "Can this really be a free string? 'klouda' is a valid value for this?"

Reply:
I agree that more information is needed here. Due to timelines, I'd like to take this as a follow-up item. @jekloudaMSFT can you please make a list (perhaps open a WI tracking input parameter validation) to keep track of follow up items on input parameters?

@ashishsingh2105 and @susanshen96 can you please provide more info on valid/expected values for these properties?

Reply:
CommandId is defined by the developer in their app manifest as part of the search command.